### PR TITLE
feat(anniversary): Add anniversary claiming modal

### PR DESCRIPTION
**Pull request by 0xCorgi**
Add modal for claiming anniversary points.

### CODE REVIEW

File: src/views/Collectibles/components/AnniversaryAchievement.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  }, [])
+
+  return (
+    <Modal title={t('Congratulations!')} onDismiss={onDismiss}>
+      <Flex flexDirection="column" alignItems="center" justifyContent="center">
+        <Box>
+          <AnniversaryImage src="/images/achievements/1-year.svg" />
+        </Box>
+        <Text textAlign="center" bold color="secondary" fontSize="24px" mb="24px">
+          {t('You won points for joigning PancakeSwap during the first year of our journey!')}
```
Comment (hachiojidev) [inline src/views/Collectibles/components/AnniversaryAchievement.tsx:55]: Typo here "joining"

File: src/views/Collectibles/components/GlobalCheckClaimStatus.tsx [hunk 2]
```diff
@@ -20,10 +22,13 @@ interface GlobalCheckClaimStatusProps {
 const GlobalCheckClaimStatus: React.FC<GlobalCheckClaimStatusProps> = ({ excludeLocations }) => {
   const hasDisplayedModal = useRef(false)
   const [claimableNfts, setClaimableNfts] = useState<Nft[]>([])
+  const [canClaimAnniversaryPoints, setCanClaimAnniversaryPoints] = useState<boolean>(false)
```
Comment (hachiojidev) [inline src/views/Collectibles/components/GlobalCheckClaimStatus.tsx:25]: You don't need to type this as `boolean`

File: src/views/Collectibles/components/AnniversaryAchievement.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  return (
+    <Modal title={t('Congratulations!')} onDismiss={onDismiss}>
+      <Flex flexDirection="column" alignItems="center" justifyContent="center">
+        <Box>
+          <AnniversaryImage src="/images/achievements/1-year.svg" />
+        </Box>
+        <Text textAlign="center" bold color="secondary" fontSize="24px" mb="24px">
+          {t('You won points for joining PancakeSwap during the first year of our journey!')}
+        </Text>
+        <Button disabled={isLoading} onClick={handleClick}>
```
Comment (ChefHutch) [inline src/views/Collectibles/components/AnniversaryAchievement.tsx:61]: Not important but you can use an `endIcon` prop to add a loading icon conditionally here too 

https://github.com/pancakeswap/pancake-frontend/blob/303b3fda6dd72551116becfeda9b9ee9fc63acb1/src/views/Lottery/components/BuyTicketsModal/EditNumbersModal.tsx#L72

File: src/views/Collectibles/components/AnniversaryAchievement.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      y: 0.3,
+    },
+  })
+}
+
+interface AnniversaryModalProps extends InjectedModalProps {
+  onClick: () => Promise<void>
+}
+
+const AnniversaryAchievementModal: React.FC<AnniversaryModalProps> = ({ onDismiss, onClick }) => {
```
Comment (ChefHutch) [inline src/views/Collectibles/components/AnniversaryAchievement.tsx:34]: Let's add `Modal` to the file name too?

File: src/views/Collectibles/components/GlobalCheckClaimStatus.tsx [hunk 5]
```diff
@@ -55,10 +60,16 @@ const GlobalCheckClaimStatus: React.FC<GlobalCheckClaimStatusProps> = ({ exclude
...
+      const canClaimAnniversary = await canClaim(account)
+      setCanClaimAnniversaryPoints(canClaimAnniversary)
+    }
+
     if (account) {
       fetchClaimStatus()
+      fetchClaimAnniversaryStatus()
     }
-  }, [account, canClaimBaller, canClaimLottie, canClaimLucky])
+  }, [account, canClaimBaller, canClaimLottie, canClaimLucky, canClaim])
```
Comment (ChefHutch) [inline src/views/Collectibles/components/GlobalCheckClaimStatus.tsx:72]: Also not too important but I think all of the claim conditions for lottery NFTs will now be falsy- so these can be removed. Nyan said that the other day but maybe double-check to be sure

**Review by ChefHutch — COMMENTED** [review]
Nice one man, couple of small comments.

You can also amend the `excludeLocations={['/collectibles']}` prop in `App.tsx` to just receive an empty array - we'll want this one to fire everywhere (that was used to prevent the modal firing on the _actual_ claim page - /collectibles. Because the claim happens within the modal itself, it should fire everywhere.

File: src/views/Collectibles/components/AnniversaryAchievement.tsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+  const handleClick = async () => {
+    setIsLoading(true)
+    try {
+      await onClick()
+    } catch (e) {
+      onDismiss()
+    }
+    onDismiss()
+  }
```
Comment (hachiojidev) [inline src/views/Collectibles/components/AnniversaryAchievement.tsx:46]: Small thing but if we are not doing anything with the error we can simply do this
```
try {
  await onClick()
} finally {
  onDismiss()
}

File: src/views/Collectibles/components/GlobalCheckClaimStatus.tsx (hunk 5 above)
Comment (RabbitDoge) [reply src/views/Collectibles/components/GlobalCheckClaimStatus.tsx:72]: yep there is a time limit on the contract

Comment (0xCorgi) [reply src/views/Collectibles/components/GlobalCheckClaimStatus.tsx]: done here 6389cae8dfad153b5863930b92f2748adfaac6d9

**Review by RabbitDoge — APPROVED** [review]

**Pull request merged**